File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
 //collect correct input
    float f;
    {
    do
    {
        f = get_float("Change owed: ");
        
    }
    while (f <= 0);
    }
//convert to cents as needed
    int cents = round(f*100);
      int q;
      int d;
      int n;
      int p;
      
//do math for quarters, x is left over coins
    q = cents/25;
    cents = cents%25;
        
//do math for dimes
    d = cents/10;
    cents = cents%10;
        
//do math for nickels
    n = cents/5;
    cents = cents%5;
        
//do math for pennies
    p = cents/1;
    
printf("%i\n", q+d+n+p);
    
} 
   
    

    
        
    

File: cash.py
from cs50 import get_float

# Prompt user for change owed
while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

# Convert to total cents owed
cents = round(change*100)

# Get amount of quarters then update cents left
q = cents // 25
cents = cents % 25

# Get amount of dimes then update cents left
d = cents // 10
cents = cents % 10

# Get amount of nickels then update cents left
n = cents // 5
cents = cents % 5

# Get amount of pennies
p = cents

# Give minimum number of coins
print(f"{(q+d+n+p)}")
